harness/determined/tensorboard/build.py

```diff
@@ -27,18 +27,18 @@ def get_rank_if_horovod_process_else_return_zero() -> Optional[int]:
     return int(os.getenv("HOROVOD_RANK", 0))
 
 
-def get_base_path(checkpoint_config: Dict[str, Any], manager: bool = False) -> pathlib.Path:
+def get_base_path(checkpoint_config: Dict[str, Any]) -> pathlib.Path:
+    allocation_id = os.environ.get("DET_ALLOCATION_ID", "")
     rank = get_rank_if_horovod_process_else_return_zero()
 
     if checkpoint_config.get("base_path"):
-        return pathlib.Path(checkpoint_config["base_path"]).joinpath("tensorboard")
```

Comment:
I wonder whether the real cause of the collision troubles we are trying to fix is this line, i. e. the fact that if the user specified a `base_path` we will not differentiate by rank.

Reply:
If we introduce the differentiation of path by rank then the only way there will be collisions is if the user runs multiple trials with _the same checkpoint storage_. In other words, if we don't introduce the allocation_id suffix but do introduce the rank ending, and they want to avoid collisions they still can if only they run each trial with a different checkpoint storage in the experiment configuration. Now that I typed this, I feel that it is not a reasonable requirement so I guess we still should have an allocation_id suffix.

Reply:
No, I don't know how long ago base_path was part of the checkpoint storage config, but it's not allowed anymore (see `schemas/expconf/v0/checkpoint-storage.json`)

It's probably time to just remove this check.